agent/structs/structs.go

```diff
@@ -1031,6 +1031,12 @@ func (ns *NodeService) CompoundServiceName() ServiceName {
 	}
 }
 
+// UniqueID is a unique identifier for a service instance within a datacenter by encoding:
+// node/namespace/service_id
```

Comment:
I thought there are very few limits to the characters allowed in a Node name. What if someone put slashes in their node names?

Reply:
For services and namespaces there's an implicit assumption that they are dns component restricted already (for Connect) given that we ultimately stuff them into SNI components and things work. For node names we don't have any particular secret validation like that IIRC.

Reply:
I don't think there would be a problem, since we wouldn't split an ID up into its components. I added a comment to say not to do that.

Reply:
:+1: